src/Microsoft.VisualBasic.Core/tests/ByteTypeTests.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.VisualBasic.Tests;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.VisualBasic.CompilerServices.Tests
+{
+    public class ByteTypeTests
+    {
+        [Theory]
+        [MemberData(nameof(FromObject_TestData))]
+        [MemberData(nameof(FromString_TestData))]
+        public void FromObject(object value, byte expected)
+        {
+            Assert.Equal(expected, ByteType.FromObject(value));
+        }
+
+        // The following conversions are not supported.
+        [Theory]
+        [MemberData(nameof(FromObject_NotSupported_TestData))]
+        public void FromObject_NotSupported(object value, byte expected)
+        {
+            Assert.Throws<InvalidCastException>(() => ByteType.FromObject(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(FromObject_Invalid_TestData))]
+        public void FromObject_ThrowsInvalidCastException(object value)
+        {
+            Assert.Throws<InvalidCastException>(() => ByteType.FromObject(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(FromObject_Overflow_TestData))]
+        public void FromObject_ThrowsOverflowException(object value)
+        {
+            Assert.Throws<OverflowException>(() => ByteType.FromObject(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(FromString_TestData))]
+        public void FromString(string value, byte expected)
+        {
+            Assert.Equal(expected, ByteType.FromString(value));
+        }
+
+        // The following conversions are not supported.
+        [Theory]
+        [MemberData(nameof(FromString_NotSupported_TestData))]
+        public void FromString_NotSupported(string value, byte expected)
+        {
+            Assert.Throws<InvalidCastException>(() => ByteType.FromString(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(FromString_Invalid_TestData))]
+        public void FromString_ThrowsInvalidCastException(string value)
+        {
+            Assert.Throws<InvalidCastException>(() => ByteType.FromString(value));
+        }
+
+        [Theory]
+        [MemberData(nameof(FromString_Overflow_TestData))]
+        public void FromString_ThrowsOverflowException(string value)
+        {
+            Assert.Throws<OverflowException>(() => ByteType.FromString(value));
+        }
+
+        private static IEnumerable<object[]> FromObject_TestData()
```

Comment:
Worth having some tests for values boxed as `object`?

Reply:
Nevermind, I'm an idiot

---
In reply to: [277031493](https://github.com/dotnet/corefx/pull/37010#discussion_r277031493) [](ancestors = 277031493)